core/create_spoken_forms.py

```diff
@@ -187,23 +182,130 @@ def create_single_spoken_form(source: str):
+        # Leave completely uppercase words alone, as we can deal with them later.
+        # Otherwise normalized the rest to help with subsequent abbreviation lookups,
+        # etc.
+        if source.isupper():
+            mapped_source = source
+        else:
+            mapped_source = source.lower()
     return mapped_source
 
 
-def create_spoken_forms_from_regex(source: str, pattern: re.Pattern):
+def create_exploded_forms(spoken_forms: List[str]):
+    """Exploded common packed words into separate words"""
+    # TODO: This could be moved somewhere else, possibly seeded from something like
+    # words to replace...
+    packed_words = {"readme": "read me"}
+
+    new_spoken_forms = []
+    for line in spoken_forms:
+        exploded_form = []
+        # ex: "vm" or "usb" explodes into "V M" or "U S B"
+        if (
+            " " not in line
+            and line.islower()
+            and len(line) > 1
+            and len(line) <= EXPLODE_MAX_LEN
+        ):
+            new_spoken_forms.append(line)  # Keep a regular copy (ie: "nas")
+            new_spoken_forms.append(" ".join(line.upper()))
+        # ex: "readme" explodes into "read me"
+        else:
+            for word in line.split(" "):
+                if word in packed_words.keys():
+                    exploded_form.append(packed_words[word])
+                else:
+                    exploded_form.append(word)
+            new_spoken_forms.append(" ".join(exploded_form))
+    return new_spoken_forms
+
+
+def create_extension_forms(spoken_forms: List[str]):
+    """Add extension forms"""
+    new_spoken_forms = []
+
+    file_extensions_map = {v.strip(): k for k, v in file_extensions.items()}
+    for line in spoken_forms:
+        have_file_extension = False
+        file_extension_forms = []
+        dotted_extension_form = []
+        truncated_forms = []
+        for substring in line.split(" "):
+            # NOTE: If we ever run in to file extensions in the middle of file name, the
+            # truncated form is going to be busted. ie: foo.md.disabled
+
+            if substring in file_extensions_map.keys():
+                file_extension_forms.append(file_extensions_map[substring])
+                dotted_extension_form.append(REVERSE_PRONUNCIATION_MAP["."])
+                dotted_extension_form.append(file_extensions_map[substring])
+                have_file_extension = True
+                # purposefully down update truncated
+            else:
+                file_extension_forms.append(substring)
+                dotted_extension_form.append(substring)
+                truncated_forms.append(substring)
+        # print(file_extension_forms)
+        if have_file_extension:
+            new_spoken_forms.append(" ".join(file_extension_forms))
+            new_spoken_forms.append(" ".join(dotted_extension_form))
+        new_spoken_forms.append(" ".join(truncated_forms))
+
+    return list(dict.fromkeys(new_spoken_forms))
```

Comment:
Is this intended to just be a `set`?

Reply:
Yep, will update when I implement the tests. Thanks.

Reply:
I updated them all to sets now